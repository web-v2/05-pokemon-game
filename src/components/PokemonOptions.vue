<template>
  <div class="options-container">
    <ul v-if="mostrar">
        <li v-for="pk in pokemons" :key="pk.id" 
        v-on:click="inactivarOpciones"         
        @click="$emit('selectionPokemon', pk.id)">{{ pk.name }}</li>
    </ul>
  </div>
</template>

<script>
export default {
    props: {
        pokemons: {
            type: Array,
            required: true
        }
    },
    data() {
        return {
            mostrar: true
        }
    },
    methods:{
         inactivarOpciones(){
            this.mostrar = false;
        }
    }
}
</script>

<style scoped>
/* Pokemon Options */
ul {
    list-style-type: none;
}
li {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    margin-bottom: 12px;
    width: 190px;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
    display: flex;
    justify-content: center;
}
</style>