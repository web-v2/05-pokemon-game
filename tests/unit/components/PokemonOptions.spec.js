import {shallowMount} from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'

import {pokemonsMock} from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {
    let wrapper
    beforeEach(() => {
        wrapper  = shallowMount(PokemonOptions, {
            props: {
                pokemons: pokemonsMock
            }
        })
    });        
    test('Debe hacer match con el snapshot ', () => {
        expect( wrapper.html() ).toMatchSnapshot()        
    });
    test('Debe de mostrar las 4 opciones correctamente', () => {
        const liTags = wrapper.findAll('li')
        expect( liTags.length ).toBe(4)

        expect( liTags[0].text() ).toBe('bulbasaur')
        expect( liTags[1].text() ).toBe('ivysaur')
        expect( liTags[2].text() ).toBe('venusaur')
        expect( liTags[3].text() ).toBe('charmander')

    });
    test('Debe emitir "selectionPokemon" con sus respectivos parametros al hacer click', () => {
        const [l1,l2,l3,l4] = wrapper.findAll('li')
        l1.trigger('click')
        l2.trigger('click')
        l3.trigger('click')
        l4.trigger('click')

        expect(wrapper.emitted('selectionPokemon').length).toBe(4)
        expect(wrapper.emitted('selectionPokemon')[0]).toEqual([1])
        expect(wrapper.emitted('selectionPokemon')[1]).toEqual([2])
        expect(wrapper.emitted('selectionPokemon')[2]).toEqual([3])
        expect(wrapper.emitted('selectionPokemon')[3]).toEqual([4])
    });
});