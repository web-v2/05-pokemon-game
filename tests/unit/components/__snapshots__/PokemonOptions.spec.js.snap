// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component Debe hacer match con el snapshot  1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
